cmd/export_orderbooks.go

```diff
@@ -0,0 +1,151 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/stellar-etl/internal/input"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+// exportOrderbooksCmd represents the exportOrderbooks command
+var exportOrderbooksCmd = &cobra.Command{
+	Use:   "export_orderbooks",
+	Short: "This command exports the historical orderbooks",
+	Long: `This command instantiates a stellar-core instance and uses it to export normalized orderbooks.
+	The information is exported in batches determined by the batch-size flag. The normalized data is exported in multiple 
+	different files within the exported data folder. These files are dimAccounts.txt, dimOffers.txt, dimMarkets.txt, and factEvents.txt.
+	These files contain normalized data that helps save storage space. 
+	
+	If the end-ledger is omitted, then the stellar-core node will continue running and exporting information as new ledgers are 
+	confirmed by the Stellar network. In this unbounded case, a stellar-core config file is required.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		endNum, useStdout, strictExport := utils.MustCommonFlags(cmd.Flags(), cmdLogger)
+
+		execPath, configPath, startNum, batchSize, outputFolder := utils.MustCoreFlags(cmd.Flags(), cmdLogger)
+		var folderPath string
+		if !useStdout {
+			folderPath = mustCreateFolder(outputFolder)
+		}
+
+		if batchSize <= 0 {
+			cmdLogger.Fatalf("batch-size (%d) must be greater than 0", batchSize)
+		}
+
+		if configPath == "" && endNum == 0 {
+			cmdLogger.Fatal("stellar-core needs a config file path when exporting ledgers continuously (endNum = 0)")
+		}
+
+		var err error
+		execPath, err = filepath.Abs(execPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for stellar-core executable: ", err)
+		}
+
+		configPath, err = filepath.Abs(configPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for the config file: ", err)
+		}
+
+		checkpointSeq := utils.GetMostRecentCheckpoint(startNum)
+		core, err := input.PrepareCaptiveCore(execPath, configPath, checkpointSeq, endNum)
+		if err != nil {
+			cmdLogger.Fatal("error creating a prepared captive core instance: ", err)
+		}
+
+		orderbook, err := input.GetEntriesFromGenesis(checkpointSeq, xdr.LedgerEntryTypeOffer)
+		if err != nil {
+			cmdLogger.Fatal("could not read initial orderbook: ", err)
+		}
+
+		input.UpdateOrderbook(checkpointSeq, startNum, orderbook, core, cmdLogger)
+		orderbookChannel := make(chan input.OrderbookBatch)
+
+		go input.StreamOrderbooks(core, startNum, endNum, batchSize, orderbookChannel, orderbook, cmdLogger)
+		if endNum != 0 {
```

Comment:
It'd be useful to add a comment explaining the significance of `endNum == 0`, since it's unclear from just looking at the code.

Other than that, very clean and clear implementation of this function!